java/code/src/com/suse/manager/webui/services/SaltStateGeneratorService.java

```diff
@@ -548,26 +550,15 @@ public void createOrg(Org org) {
 
     /**
      * Expose some global configuration options as pillar data.
-     * @param saltRootPath the directory where to generate the pillar data file
      */
-    public void generateMgrConfPillar(Path saltRootPath) {
-        boolean metataSigningEnabled = ConfigDefaults.get().isMetadataSigningEnabled();
-        SaltPillar pillar = new SaltPillar();
-        pillar.add("mgr_metadata_signing_enabled", metataSigningEnabled);
-        Path filePath = saltRootPath.resolve("mgr_conf." + PILLAR_DATA_FILE_EXT);
+    public static void generateMgrConfPillar() {
+        // Remove legacy file if present
+        Path filePath = Paths.get(SUMA_PILLAR_DATA_PATH).resolve("mgr_conf." + PILLAR_DATA_FILE_EXT);
         FileUtils.deleteFile(filePath);
-        try {
-            com.suse.manager.webui.utils.SaltStateGenerator saltStateGenerator =
-                    new com.suse.manager.webui.utils.SaltStateGenerator(filePath.toFile());
-            saltStateGenerator.generate(pillar);
-            if (!skipSetOwner) {
-                FileUtils.setAttributes(filePath, "tomcat", "susemanager",
-                        Set.of(OWNER_READ, OWNER_WRITE, GROUP_READ, GROUP_WRITE));
-            }
-        }
-        catch (IOException e) {
-            LOG.error("Failed to generate pillar data into " + filePath, e);
-            throw new RuntimeException(e);
+
+        boolean metataSigningEnabled = ConfigDefaults.get().isMetadataSigningEnabled();
+        if (Pillar.getGlobalPillars().stream().noneMatch(pillar -> pillar.getCategory().equals("mgr_conf"))) {
+            Pillar.createGlobalPillar("mgr_conf", Map.of("mgr_metadata_signing_enabled", metataSigningEnabled));
```

Comment:
Do I understand it right that this does not regenerate mgr_conf pillar once it exists? Do we want this behavior? With the files it was possible to regenerate.

Reply:
hum... right. I wrongly assumed that this was never changed. I'll fix that.